src/actions/removeMulticursor.ts

```diff
@@ -2,15 +2,18 @@ import _ from 'lodash'
 import Path from '../@types/Path'
 import State from '../@types/State'
 import Thunk from '../@types/Thunk'
+import expandThoughts from '../selectors/expandThoughts'
 import hashPath from '../util/hashPath'
 
-/** Removes a cursor from the multicursor set. */
+/** Removes a cursor from the multicursor set and updates expanded state. */
 const removeMulticursor = (state: State, { path }: { path: Path }): State => {
   const { [hashPath(path)]: _, ...remainingMulticursors } = state.multicursors
 
   return {
     ...state,
     multicursors: remainingMulticursors,
+    // Update expanded state based on remaining cursor or multicursors
+    expanded: expandThoughts(state, state.cursor),
```

Comment:
`state` is the old state before the multicursor is removed. You need to pass it an up-to-date state.